### ISSUE TITLE
double_parens: consider one diagnostic instead of one per pair?

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#17475]
### Description

```rust
#![allow(unused_parens)]
#![allow(non_upper_case_globals)]
#![allow(dead_code)]
#![cfg_attr(rustfmt, rustfmt::skip)]
static a: isize = ((((1))));

pub fn main() {}
```

clippy will print
````
warning: unnecessary parentheses
 --> a.rs:5:19
  |
5 | static a: isize = ((((1))));
  |                   ^^^^^^^^^ help: remove them: `(((1)))`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#double_parens
  = note: `#[warn(clippy::double_parens)]` on by default

warning: unnecessary parentheses
 --> a.rs:5:20
  |
5 | static a: isize = ((((1))));
  |                    ^^^^^^^ help: remove them: `((1))`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#double_parens

warning: unnecessary parentheses
 --> a.rs:5:21
  |
5 | static a: isize = ((((1))));
  |                     ^^^^^ help: remove them: `(1)`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#double_parens

warning: 3 warnings emitted
````

which seems a bit over the top, can clippy figure out that we can reduce `((((1))))` to `(1)` in one go?

```
warning: unnecessary parentheses
 --> a.rs:5:19
  |
5 | static a: isize = ((((1))));
  |                   ^^^^^^^^^ help: remove them: `(1)`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#double_parens
  = note: `#[warn(clippy::double_parens)]` on by default
```

### Version

```text
rustc 1.99.0-nightly (1a833e165 2026-07-29)
binary: rustc
commit-hash: 1a833e16546c2eb012758ddd499964fd8afee29e
commit-date: 2026-07-29
host: x86_64-unknown-linux-gnu
release: 1.99.0-nightly
LLVM version: 22.1.8
```

**Comment by Jarcho** [comment]
Does this ever actually happen in practice? It's not in any way hard to handle, but it requires a skip counter and that makes the lint code a little harder to reason about.

**Comment by lsunsi** [comment]
If this is something that we want to work towards, I'd be willing to give a try.

**Comment by dswij** [comment]
> Does this ever actually happen in practice? 

rustfmt already formats this, so I doubt that it's actually a common issue.
